### DISCUSSION TITLE
About Cockpit's Multilingual

### DISCUSSION THREAD

**Discussion by zhaojing1987** [cockpit-project/cockpit#20688] [Q&A]
I want to know if there is a way to set the default language in cockpit, which is similar to a configuration file, such as zh_CN. I don't seem to find a way. Cockpit seems to automatically select the default language based on the browser.
My problem now is that my msgid is similar to a variable. I translated it into en_US and zh_CN, but when I set the browser language to a language other than these two, it shows msgid, which is not what I need. I know there is a problem with this method, but it seems difficult to modify it now. Is there any solution?
![image](https://github.com/cockpit-project/cockpit/assets/7624828/2ca430dd-4978-45f8-b8c1-c3b8703a3642)
![image](https://github.com/cockpit-project/cockpit/assets/7624828/eb1c9c78-0ed9-41f5-b8cc-9f83023e0c40)

**Comment by martinpitt** [comment]
You can change the language in the "Session" menu at the top right. That setting will be stored as a cookie in your browser, that's about as "similar to a configuration file" as you can get with a browser application.

> My problem now is that my msgid is similar to a variable.

I suppose you mean on your custom cockpit page? That's certainly not what we do for the "official" cockpit pages, msgids are English there.
